Automattic/jetpack
# Add/bundles to recommendations

## Proposed changes:

* Add product cards for bundles (Security, Complete, and Growth)
* Add interstitials for bundles
* Add complete class to enable interstitial and product card for it
* Fix issue with ineligible intro-discounts showing on recommendation cards

### Other information:


## Jetpack product discussion

P2: pbNhbs-bxX-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. Make sure your JN site has access to your sandbox
3. Enable your sandbox on your testing site
4. Checkout the following diff on your sandbox: D166839-code
5. Go to My Jetpack and fill out the welcome banner survey and check all possible options so you can see Jetpack Complete
![image](https://github.com/user-attachments/assets/bda99d9c-9071-4498-a2d0-b9f274185d24)
6. Ensure you see the Jetpack Complete recommendation. Make sure the "Purchase" button goes to checkout with Complete in the cart
![image](https://github.com/user-attachments/assets/f09aa052-aeac-4a9b-a621-4b8fcc7ab5c6)
7. Click on "Learn More" to see the new Complete interstitial. Make sure the interstitial works correctly by testing the CTA and ensuring it goes to checkout with Complete in the cart.
![image](https://github.com/user-attachments/assets/eeaddfc6-2b02-480d-86f1-3de0b906520b)
![image](https://github.com/user-attachments/assets/181866b7-9f93-4497-8fdd-a8d319483fff)
8. Go back to My Jetpack and redo the recommendation survey. This time select `Protect my site` only. This should ensure you see the Security card. Test the same things you did with Complete with the interstitial and the card itself
![image](https://github.com/user-attachments/assets/af8346db-d691-42da-8708-294954fbd1a1)
![image](https://github.com/user-attachments/assets/493907ff-cc08-4063-8a0c-7a53af0297da)
9. Go back to My Jetpack one more time and redo the survey again. This time, select only `Grow my audience`. Test this the same way you tested the last two bundles
![image](https://github.com/user-attachments/assets/d260a214-8cde-459b-81f4-82101e42c010)
![image](https://github.com/user-attachments/assets/a0726ade-7411-4357-b474-e2497098b8c1)
10. You've now tested all the new interstitials and recommendation cards. Now go to `/wp-admin/admin.php?page=my-jetpack#/add-social` and ensure that the bundle upsells still look like they used to and should
![image](https://github.com/user-attachments/assets/ad75a3ec-c86e-4aaa-b68f-3a6b77b92f90)
11. Lastly, we'll test the intro discount eligibility on the cards. Purchase any plan that you see in your recommendations on the site. For this example, I'll use Protect.
12. After purchasing protect go remove it from your site via wordpress.com/me/purchases or the Store Admin
13. Go back to My Jetpack and ensure you no longer see the intro discount on the card
![image](https://github.com/user-attachments/assets/4512883d-6454-4b1f-8ff0-14d7fb42dbe9)
